src/client/Client.cc

```diff
@@ -257,6 +257,16 @@ int Client::get_fd_inode(int fd, InodeRef *in) {
   return r;
 }
 
+bool Client::fh_exists(Fh *f) {
+  if (ll_unclosed_fh_set.count(f)) {
+    return true;
+  }
+
+  return std::any_of(fd_map.begin(), fd_map.end(), [&f](const auto& it) {
+    return f == it.second;
+  });
```

Comment:
* Why is an `O(n)` loop required here ?
* Also, `f` doesn't need to be captured by reference for the lamba function since we are not modifying the pointer. So, a simple `[f]` will suffice since `f` is a pointer to a struct.

Reply:
> * Why is an `O(n)` loop required here ?

I'm looking for the value, so I have to iterate through the entire map.

> * Also, `f` doesn't need to be captured by reference for the lamba function since we are not modifying the pointer. So, a simple `[f]` will suffice since `f` is a pointer to a struct.

Ah right, makes sense, will update it

Reply:
> > * Why is an `O(n)` loop required here ?
> 
> I'm looking for the value, so I have to iterate through the entire map.

Why would the file handle pointer be found for a different file descriptor than the one that's passed in ?

> 
> > * Also, `f` doesn't need to be captured by reference for the lamba function since we are not modifying the pointer. So, a simple `[f]` will suffice since `f` is a pointer to a struct.
> 
> Ah right, makes sense, will update it

Reply:
> > > * Why is an `O(n)` loop required here ?
> > 
> > 
> > I'm looking for the value, so I have to iterate through the entire map.
> 
> Why would the file handle pointer be found for a different file descriptor than the one that's passed in ?

There's no fd passed, what im looking for is a file handle out of all the available file handles .

Reply:
`ceph::unordered_map<int, Fh*> fd_map;`

what is passed in the helper i wrote is file handle pointer and not the file descriptor

Reply:
> `ceph::unordered_map<int, Fh*> fd_map;`
> 
> what is passed in the helper i wrote is file handle pointer and not the file descriptor

I was referring to the current use case for `fh_exists()`
The current use case gets called from `_preadv_pwritev_locked()`
However, the file handle itself is extracted from `fd_map` in `get_filehandle()` called from `_preadv_pwritev()`

So, my argument is, there isn't a random function call just passing in a file handle that doesn't belong to the current file descriptor. So, you could just return `nullptr` in `fh_exists()` if the file handle can't be found in `ll_unclosed_fh_set`. This will avoid iterating through the entire map.

Reply:
hmm, my intention was to make it a generic function that can be used from anywhere (i.e. to factor in `fd_map` usages) but looking more closely it seems there's no need, the API making use of fd_map is open() and openat() which both accept fd as arg but not fh and all the I/O APIs accepting fh directly or indirectly call _preadv_pwritev() so it does make sense to remove the iterating part

Reply:
@mchangir done, PTAL